### PR TITLE
Adding resource and docs for VPN gateways

### LINKED ISSUE #330: Adding resource for Azure VPN Gateways
## Detailed Description

Adding a resource using the new backend, to enable checks on Azure VPN Gateways

## Context

This could be used to verify the presence of a VPN Gateway or specific properties of this VPN gateway.

## Possible Implementation

Adding a PR which includes code/docs for this.

### PR DESCRIPTION
**Pull request by voojoe**
Signed-off-by: Joe McCrea <joe.mccrea@sap.com>

### Description

Adding in resources and documentation for Azure VPN Gateways

### Issues Resolved

#330 

### Check List

- [x] New functionality includes tests
- [x] All tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

**Review by omerdemirok — COMMENTED** [review]
Thanks for this @voojoe!
Do you consider changing the name of the resource to `azure_virtual_network_gateways`? This will align better with the official Azure resource name [here](https://docs.microsoft.com/en-us/rest/api/network-gateway/virtualnetworkgateways/list).

File: docs/resources/azure_vpn_gateways.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+## Availability
+
+### Installation
+
+This resource is available in the [InSpec Azure resource pack](https://github.com/inspec/inspec-azure). 
+For an example `inspec.yml` file and how to set up your Azure credentials, refer to resource pack [README](../../README.md#Service-Principal).
+
+## Syntax
+
+An `azure_vpn_gateways` resource block returns all VPN gateways, either within a Resource Group (if provided), or within an entire Subscription.
```
Comment (omerdemirok) [inline docs/resources/azure_vpn_gateways.md:29]: Seems like the `resource_group` is a mandatory URI parameter, [here](https://docs.microsoft.com/en-us/rest/api/network-gateway/virtualnetworkgateways/list#uri-parameters).

File: libraries/azure_vpn_gateways.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    # Define the column and field names for FilterTable.
+    # In most cases, the `column` should be the pluralized form of the `field`.
+    # @see https://github.com/inspec/inspec/blob/master/docs/dev/filtertable-usage.md
+    table_schema = [
+      { column: :names, field: :name },
+      { column: :ids, field: :id },
+      { column: :types, field: :type },
+      { column: :locations, field: :location },
+      { column: :tags, field: :tag },
```
Comment (omerdemirok) [inline libraries/azure_vpn_gateways.rb:35]: This should be `{ column: :tags, field: :tags },`. Both `tags` are plural. See, [here](https://docs.microsoft.com/en-us/rest/api/network-gateway/virtualnetworkgateways/list#virtualnetworkgateway).

File: libraries/azure_vpn_gateways.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  EXAMPLE
+
+  attr_reader :table
+
+  def initialize(opts = {})
+    # Options should be Hash type. Otherwise Ruby will raise an error when we try to access the keys.
+    raise ArgumentError, 'Parameters must be provided in an Hash object.' unless opts.is_a?(Hash)
+
+    opts[:resource_provider] = specific_resource_constraint('Microsoft.Network/virtualNetworkGateways', opts)
+
```
Comment (omerdemirok) [inline libraries/azure_vpn_gateways.rb:19]: `resource_group` should be mandatory. See, [resource creation guideline](https://github.com/inspec/inspec-azure/blob/master/docs/resource_creation_guide.md#required-parameters) and an example usage [here](https://github.com/inspec/inspec-azure/blob/master/libraries/azure_mysql_databases.rb#L20).

File: libraries/azure_vpn_gateways.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
+require 'azure_generic_resources'
+
+class AzureVPNGateways < AzureGenericResources
+  name 'azure_vpn_gateways'
+  desc 'Verifies settings for AKS VPN Gateways'
```
Comment (omerdemirok) [inline libraries/azure_vpn_gateways.rb:5]: Looks like there is a typo here: Azure Kubernetes Service (AKS).

**Review by omerdemirok — COMMENTED** [review]
This resource should be added to the project README.

File: docs/resources/azure_vpn_gateways.md (hunk 1 above)
Comment (voojoe) [reply docs/resources/azure_vpn_gateways.md]: Hi Omer, I've renamed that now in my latest commit.

**Comment by voojoe** [comment]
> Thanks for this @voojoe!
> Do you consider changing the name of the resource to `azure_virtual_network_gateways`? This will align better with the official Azure resource name [here](https://docs.microsoft.com/en-us/rest/api/network-gateway/virtualnetworkgateways/list).

Hi Omer, I've renamed this now in my latest commit.

File: docs/resources/azure_vpn_gateways.md (hunk 1 above)
Comment (voojoe) [reply docs/resources/azure_vpn_gateways.md]: Upon testing this it doesn't seem to be mandatory, I can call it with and without a resource group and it seems to behave as expected.

File: libraries/azure_vpn_gateways.rb (hunk 2 above)
Comment (voojoe) [reply libraries/azure_vpn_gateways.rb]: I've amended this now, thanks.

File: libraries/azure_vpn_gateways.rb (hunk 4 above)
Comment (voojoe) [reply libraries/azure_vpn_gateways.rb]: I've changed this now, thanks for spotting this.

File: libraries/azure_vpn_gateways.rb (hunk 3 above)
Comment (voojoe) [reply libraries/azure_vpn_gateways.rb]: It seems to work as an optional parameter when I've been testing it and behaving as expected.

File: docs/resources/azure_vpn_gateways.md (hunk 1 above)
Comment (omerdemirok) [reply docs/resources/azure_vpn_gateways.md]: Perfect, thanks for the clarification!

**Review by omerdemirok — APPROVED** [review]
Thanks for this excellent contribution @voojoe!!
It is a milestone for being the first community resource with the new backend.

**Pull request merged**